src/Components/Components/src/DynamicComponent.cs

```diff
@@ -0,0 +1,104 @@
+        [Parameter]
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
+        public Type Type { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets a dictionary of parameters to be passed to the component.
+        /// </summary>
+        // Note that this deliberately does *not* use CaptureUnmatchedValues. Reasons:
+        // [1] The primary scenario for DynamicComponent is where the call site doesn't
+        //     know which child component it's rendering, so it typically won't know what
+        //     set of parameters to pass either, hence the developer most likely wants to
+        //     pass a dictionary rather than having a fixed set of parameter names in markup.
+        // [2] If we did have CaptureUnmatchedValues here, then it would become a breaking
+        //     change to ever add more parameters to DynamicComponent itself in the future,
+        //     because they would shadow any coincidentally same-named ones on the target
+        //     component. This could lead to application bugs.
+        [Parameter]
+        public IDictionary<string, object>? Parameters { get; set; }
+
+        /// <inheritdoc />
+        public void Attach(RenderHandle renderHandle)
+        {
+            _renderHandle = renderHandle;
+        }
+
+        /// <inheritdoc />
+        public Task SetParametersAsync(ParameterView parameters)
+        {
+            // This manual parameter assignment logic will be marginally faster than calling
+            // ComponentProperties.SetProperties.
+            foreach (var entry in parameters)
+            {
+                if (entry.Name.Equals(nameof(Type), StringComparison.OrdinalIgnoreCase))
+                {
+                    Type = (Type)entry.Value;
+                }
+                else if (entry.Name.Equals(nameof(Parameters), StringComparison.OrdinalIgnoreCase))
+                {
+                    Parameters = (IDictionary<string, object>)entry.Value;
+                }
+                else
+                {
+                    throw new InvalidOperationException(
+                        $"{nameof(DynamicComponent)} does not accept a parameter with the name '{entry.Name}'. To pass parameters to the dynamically-rendered component, use the '{nameof(Parameters)}' parameter.");
+                }
+            }
+
+            _renderHandle.Render(_cachedRenderFragment);
+            return Task.CompletedTask;
+        }
+
+        void Render(RenderTreeBuilder builder)
+        {
+            if (Type is null)
+            {
+                throw new InvalidOperationException($"{nameof(DynamicComponent)} requires a non-null value for the parameter {nameof(Type)}.");
+            }
+
+            builder.OpenComponent(0, Type);
+
+            if (Parameters != null)
+            {
+                foreach (var entry in Parameters)
+                {
+                    builder.AddAttribute(1, entry.Key, entry.Value);
```

Comment:
What about using `builder.AddMultipleAttributes` here? It looks like it handles multiple attributes with the same name in a case insensitive way. https://github.com/dotnet/aspnetcore/blob/66d192435d69d62e8c2deee93ea2b541c891d33f/src/Components/Components/src/Rendering/RenderTreeBuilder.cs#L720-L727
Or is this specifically used for `CaptureUnmatchedValues = true` parameters?

Reply:
`AddMultipleAttributes` is useful if you're combining a dictionary of parameters before or after other parameters, as it deals with detecting duplicates and handling precedence rules.

In this case there are no other parameters - there's just the dictionary - so there's no reason to pay the extra cost of checking for duplicates.